ilp-node/src/main/java/org/interledger/node/channels/Channel.java

```diff
@@ -0,0 +1,182 @@
+import org.interledger.node.handlers.ChannelOpenedEventHandler;
+import org.interledger.node.handlers.IncomingRequestHandler;
+import org.interledger.node.handlers.IncomingTransferEventHandler;
+
+import java.io.Closeable;
+import java.util.concurrent.Future;
+
+/**
+ * Defines an abstraction that is meant to be plugged-in to an ILP node system in order
+ * for it to communicate with a another node.
+ *
+ * <p>Based on IL-RFC-24, the general contract for a channels is that, for a single account, it can
+ * send ILP packets and transfers from to the destination node all of which impact the an account
+ * linked to this channel.
+ *
+ * <p>Additionally, channel plugins can also emit events received from the peer, such as incoming
+ * ILP packets or transfers.
+ *
+ * <p>An ILP node will have a unique instance of a channels for each account.
+ *
+ * <p>The following high-level component diagram illustrates how ledger plugins are used by a
+ * connector:
+ *
+ * @see "https://interledger.org/rfcs/0024-ledger-plugin-interface-2/"
+ */
+public interface Channel extends Closeable {
+
+  /**
+   * Called to open the channel.
+   *
+   * <p>This should initiate event subscriptions and establish a connection to the peer.
+   *
+   * <p>Once this method is called, the channels MUST attempt to subscribe to and report events,
+   * including the "open" event immediately after a successful connection is established. If the
+   * connection is lost, the ledger channels SHOULD emit the "closed" event.
+   *
+   */
+  void open();
+
+  /**
+   * Called to disconnect this channel from the peer.
+   *
+   * @throws ChannelNotOpenException if the channels is not connected.
+   *
+   */
+  void close();
+
+  /**
+   * Query whether the channel is currently open.
+   *
+   * @return {@code true} if the channel is open, {@code false} otherwise.
+   */
+  boolean isOpen();
+
+  /**
+   * Initiate a transfer of money to the peer.
+   *
+   * <p>The ILP node will reduce the balance on the account linked to this channels if this call
+   * completes successfully.
+   *
+   * <p>Example: If this channels is used to make transfers over a payment channel then this should
+   * result in a new claim being sent over the channel.
+   *
+   * @param amount The amount to credit the account in favour of the sender.
+   *
+   * @throws ChannelNotOpenException if the channel is not open.
+   *
+   * @return a Future that resolves when the transfer has been sent.
+   */
+  Future<Void> sendTransfer(long amount);
+
+  /**
+   * Sends an ILP request packet to the peer and returns the response packet.
+   *
+   * @param request The request to send to the peer.
+   *
+   * @throws ChannelNotOpenException if the channels is not connected.
+   * @throws InterledgerProtocolException if the request is rejected by the peer.
+   *
+   * @return A Future that resolves to the ILP response from the peer.
+   */
+  Future<InterledgerFulfillPacket> sendRequest(InterledgerPreparePacket request);
```

Comment:
I think I prefer `sendPacket` instead -- it's surprising to see a method called "sendRequest" and then see that I don't actually do anything with an object that refers to a "request" anywhere.

Reply:
This is interesting because when I discussed this with @emschwartz we concluded that an ILP Prepare is a request and an ILP Fulfill is a response.

The LPIv2 model is to treat this as raw data and the plugins assume it can be serialized/deserialized into an ILP packet.

I have a thought to wrapping this in a Request/Response which for now might only have the ILP packet but in future might contain the raw request bytes to allow the nodes to have very fast in-place replacement of data rather than a deserialization and then serialization step.

Reply:
Makes sense - this is fine for now - maybe in future have two methods - one that takes the packet (like you have now) and one takes a request with bytes?